app/src/ereferrals/java/org/eyeseetea/malariacare/domain/boundary/repositories/ICurrentLanguageRepository.java

```diff
@@ -0,0 +1,7 @@
+package org.eyeseetea.malariacare.domain.boundary.repositories;
+
+import org.eyeseetea.malariacare.data.IDataSourceCallback;
+
+public interface ICurrentLanguageRepository {
+    void getCurrentLanguage(IDataSourceCallback<String> callback);
```

Comment:
This method retrieves data from preferences and is not necessary return data invoking callback, can you refactor for return domain entity directly?